src/pybind/mgr/volumes/fs/operations/versions/subvolume_base.py

```diff
@@ -371,7 +371,37 @@ def set_attrs(self, path, attrs):
                 raise VolumeException(-errno.EINVAL,
                                       "invalid fscrypt_file specified: '{0}'".format(fscrypt_file))
 
+    def separate_unit_suffix(self, subvol_size):
+        import re
+        unit_map = {"B": 1,
+                    "K": pow(1024, 1),
+                    "M": pow(1024, 2),
+                    "G": pow(1024, 3),
+                    "T": pow(1024, 4),
+                    "P": pow(1024, 5),
+                    "E": pow(1024, 6)}
+        pattern = r'^(\d+\.\d+|\d+)([KMGTPE]i?B?|B)$'
+        match = re.match(pattern, subvol_size)
+        if match:
+            value = match.group(1)
+            unit = match.group(2)
+            if unit[0] == "B" and "." in value:
+                raise VolumeException(-errno.EINVAL,
+                                      f"Invalid byte value: {value}")
+            multiplier = unit_map.get(unit[0])
+            if multiplier is not None:
+                return float(value) * multiplier
+            else:
+                raise VolumeException(-errno.EINVAL,
+                                      f"Invalid subvolume unit: {unit}")
+        else:
+            return None
```

Comment:
I believe it's impossible for execution to reach here, so this can be removed.

Reply:
for very weird inputs like Ki10 or something there will be no match. check `test_subvolume_resize_invalid_unit()`.